src/hotspot/share/opto/library_call.cpp

```diff
@@ -5489,52 +5489,68 @@ void LibraryCallKit::create_new_uncommon_trap(CallStaticJavaNode* uncommon_trap_
   uncommon_trap_call->set_req(0, top()); // not used anymore, kill it
 }
 
-//------------------------------inline_array_partition-----------------------
-bool LibraryCallKit::inline_array_partition() {
-
-  Node* elementType     = argument(0);
-  Node* obj             = argument(1);
-  Node* offset          = argument(2);
-  Node* fromIndex       = argument(4);
-  Node* toIndex         = argument(5);
-  Node* indexPivot1     = argument(6);
-  Node* indexPivot2     = argument(7);
-
-  Node* pivotIndices = nullptr;
-
+// Common checks for array sorting intrinsics arguments.
+// Returns `true` if checks passed.
+bool LibraryCallKit::check_array_sort_arguments(Node* elementType, Node* obj, BasicType* bt) {
```

Comment:
Thanks for doing the update, looks good! Just one small detail: You could use a reference here instead of a pointer. Then you can pass `bt` in and modify it directly.

Reply:
Done.